stix/idl/processing/imaging/stx_estimate_flare_location.pro

```diff
@@ -0,0 +1,121 @@
+;+
+;
+; NAME:
+;   stx_estimate_flare_location
+;
+; PURPOSE:
+;   Automatic estimate of the flare location (arcsec, Helioprojective Cartesian coordinates, Solar Orbiter vantange point) 
+;
+; CALLING SEQUENCE:
+;   stx_estimate_flare_location, path_sci_file, time_range, aux_data, flare_loc=flare_loc
+;
+; INPUTS:
+;   path_sci_file: string containing the path of the science L1 fits file of the event
+;   
+;   time_range: string array containing the selected start and the end time 
+;   
+;   aux_data: auxiliary data structure corresponding to the selected time range 
+;   
+;   
+;
+; KEYWORDS:
+;   flare_loc: output, estimate of the flare location (arcsec, Helioprojective Cartesian coordinates, 
+;                      Solar Orbiter vantange point)
+;                 
+;   path_bkg_file: path of the background L1 fits file used for background subtraction
+;   
+;   energy_range: energy range to be considered for performing the full-disk Back Projection and the forward fitting method 
+;                 (default, 6-10 keV)
+;   
+;   imsize: number of pixels of the full-disk Back Projection map (default, [512,512])
+;   
+;   mapcenter: center of the Back Projection map (default, [0.,0.])
+;   
+;   subc_index: array containing the indices of the subcollimators used for computing the Back Projection map 
+;               (default, detectors from 7 to 10)
+;               
+;   silent: set to 1 for avoiding the plot of the full-disk the Back Projection map and of the estimated flare location 
+;           (default, 0)
+;   
+; HISTORY: October 2022, Massa P. (WKU), initial release
+;
+; CONTACT:
+;   paolo.massa@wku.edu
+;-
+
+pro stx_estimate_flare_location, path_sci_file, time_range, aux_data, flare_loc=flare_loc, path_bkg_file=path_bkg_file, $
+                                 energy_range=energy_range, imsize=imsize, mapcenter=mapcenter, $
+                                 subc_index=subc_index, silent=silent, this_win=this_win, _extra=extra
+
+default, energy_range, [6.,10.]
+default, imsize, [512,512]
+default, mapcenter, [0., 0.]
+default, subc_index, stix_label2ind(['10a','10b','10c','9a','9b','9c','8a','8b','8c','7a','7b','7c'])
+default, silent, 0
+default, this_win, 4
+
+;;******* Determine the pixel size from the apparent radius of the Sun for computing a full-disk Back Projection map
+rsun = aux_data.RSUN
+pixel = rsun * 2.6 / imsize
+
+;;******* Construct the visibility structure
+mapcenter_stix = stx_hpc2stx_coord(mapcenter, aux_data)
+
+vis = stx_construct_calibrated_visibility(path_sci_file, time_range, energy_range, mapcenter_stix, $
+                                          path_bkg_file=path_bkg_file,subc_index=subc_index,  /silent, _extra=extra)
+
+
+;;******* Use Giordano's (u,v) points: no need to perform projection correction (see Giordano et al., 2015)
+subc_str = stx_construct_subcollimator()
```

Comment:
Does this not give back normal u, v points - so not Giordano's?

Reply:
Lines 69-75 create a visibility structure and change the definition of the (u,v) points from the original ones to Giordano's ones. Giordano's (u,v)-points do not require projection correction and, hence, do not need an a priori estimate of the flare location, which is determined by this procedure.
In the future, we will use Giordano's (u,v)-points by default, but we need to discuss more on this

Reply:
Sorry my bad I see now.